src/components/logInForm/LogInForm.js

```diff
@@ -1,19 +1,73 @@
 import styles from "@/styles/logInModal.module.scss";
+import loginUser from "@/utils/loginUser";
 
-const LogInForm = ({setToRegister, setToForgotPassword, setToAnonymous}) => {
+const LogInForm = ({
+  setToRegister,
+  setToForgotPassword,
+  setToAnonymous,
+  setIsLoggedIn,
+  setDisplayPopup,
+  setPopupTitle,
+  setPopupMessage
+}) => {
+  const clickHandler = async (e) => {
+    e.preventDefault();
+    const username = document.querySelector("#user").value;
+    const password = document.querySelector("#password").value;
+    let isLoggedIn = await loginUser(username, password);
+    if (isLoggedIn == true) {
+      setIsLoggedIn(true);
+      setPopupTitle("Nu är du inloggad");
+      setPopupMessage("Välkommen tillbaka, " + username);
+    } else {
+      //kommer in hit ifall användaren inte finns eller lösenordet är fel.
+      setPopupTitle("Woops! Något gick fel!");
+      setPopupMessage(isLoggedIn);
+    }
+    setDisplayPopup(true);
+  };
   return (
     <>
       <h2>Logga in</h2>
-      <a onClick={setToRegister} className={styles.logInModalLink}>Registrera ny användare</a>
+      <a onClick={setToRegister} className={styles.logInModalLink}>
+        Registrera ny användare
+      </a>
       <form data-testid="logInForm">
-        <label className={styles.screenReader} htmlFor="user">Användarnamn</label>
-        <input id="user" name="user" required placeholder="Användarnamn" type="text"/>
-        <label className={styles.screenReader} htmlFor="password">Lösenord</label>
-        <input id="password" name="password" required placeholder="Lösenord" type="password"/>
-        <a onClick={setToForgotPassword} className={styles.forgotPasswordLink}>Glömt lösenord?</a>
-        <button data-testid="logInButton" className={styles.continueButton} type="submit">Logga in</button>
+        <label className={styles.screenReader} htmlFor="user">
+          Användarnamn
+        </label>
+        <input
+          id="user"
+          name="user"
+          required
+          placeholder="Användarnamn"
+          type="text"
+        />
+        <label className={styles.screenReader} htmlFor="password">
+          Lösenord
+        </label>
+        <input
+          id="password"
+          name="password"
+          required
+          placeholder="Lösenord"
+          type="password"
+        />
+        <a onClick={setToForgotPassword} className={styles.forgotPasswordLink}>
+          Glömt lösenord?
+        </a>
+        <button
+          data-testid="logInButton"
+          onClick={(e) => clickHandler(e)}
+          className={styles.continueButton}
+          type="submit"
+        >
+          Logga in
+        </button>
```

Comment:
You don't have to add the `onClick={(e) => clickHandler(e)}` (e) parameter here, `onClick={clickHandler}` will function exactly the same. 😎